src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt

```diff
@@ -10,7 +10,33 @@ if (GEN_SHARED_LIB)
     include (GenerateExportHeader)
 endif()
 
-if(${CLR_CMAKE_HOST_ARCH} STREQUAL x86 OR ${CLR_CMAKE_HOST_ARCH} STREQUAL x64)
+if(${CLR_CMAKE_HOST_ARCH} STREQUAL arm OR ${CLR_CMAKE_HOST_ARCH} STREQUAL arm64)
```

Comment:
This commit should probably be broken into two. One that only adds the zlib-ng files. And another that connects dotnet runtime build system to use them. It would make it easier to review because then we can see what build system changes are needed for zlib-ng. But I realize at this point it would be difficult to split it.

Reply:
Good point. I'll do that.